--- uClibc-0.9.33.2/Rules.mak~	2022-04-22 10:04:03.000000000 +0200
+++ uClibc-0.9.33.2/Rules.mak	2022-04-22 10:08:16.933174949 +0200
@@ -170,7 +170,7 @@ check_as=$(shell \
 	if $(CC) -Wa,$(1) -Wa,-Z -c -o /dev/null -xassembler /dev/null > /dev/null 2>&1; \
 	then echo "-Wa,$(1)"; fi)
 check_ld=$(shell \
-	if $(LD) $(1) -o /dev/null -b binary /dev/null > /dev/null 2>&1; \
+	if $(LD) $(1) -o /dev/null -b binary /dev/zero > /dev/null 2>&1; \
 	then echo "$(1)"; fi)
 
 # Use variable indirection here so that we can have variable
