--- uClibc-0.9.28/libc/sysdeps/linux/x86_64/sigaction.c.orig	2005-08-28 10:59:32.000000000 +0200
+++ uClibc-0.9.28/libc/sysdeps/linux/x86_64/sigaction.c	2005-08-28 10:58:49.000000000 +0200
@@ -27,7 +27,7 @@
 
 
 #if defined __NR_rt_sigaction
-#warning Yes there are two warnings here.  Don't worry about it.
+#warning "Yes there are two warnings here.  Don't worry about it."
 static void restore_rt (void) asm ("__restore_rt");
 static void restore (void) asm ("__restore");
 
