diff -ru uClibc/libpthread/linuxthreads.old/Makefile.in uClibc.newsoname/libpthread/linuxthreads.old/Makefile.in
--- uClibc/libpthread/linuxthreads.old/Makefile.in	2006-01-03 09:12:10.000000000 +0100
+++ uClibc.newsoname/libpthread/linuxthreads.old/Makefile.in	2006-01-04 17:36:29.000000000 +0100
@@ -71,9 +71,14 @@
 headers-$(UCLIBC_HAS_THREADS) += linuxthreads_headers
 headers_clean-y += linuxthreads_headers_clean
 
-$(top_builddir)lib/libpthread.so: $(libpthread_OUT)/libpthread_so.a $(libc)
+$(top_builddir)lib/libpthread-uclibc.so: $(libpthread_OUT)/libpthread_so.a $(libc)
+	@# link.so takes info about soname from $@
 	$(call link.so,$(libpthread_FULL_NAME),$(MAJOR_VERSION))
 
+$(top_builddir)lib/libpthread.so: $(top_builddir)lib/libpthread-uclibc.so
+	$(LN) -sf $(firstword $^) $@.$(MAJOR_VERSION)
+	$(LN) -sf $(firstword $^) $@
+	
 $(libpthread_OUT)/libpthread_so.a: $(libpthread-so-y)
 	$(Q)$(RM) $@
 ifeq ($(PTHREADS_DEBUG_SUPPORT),y)
