--- uClibc-0.9.20/libc/sysdeps/linux/common/Makefile.orig	Wed Mar  5 11:28:53 2003
+++ uClibc-0.9.20/libc/sysdeps/linux/common/Makefile	Thu Jul 24 17:07:17 2003
@@ -39,6 +39,7 @@
 
 CTOR_TARGETS=$(TOPDIR)lib/crti.o $(TOPDIR)lib/crtn.o
 SAFECFLAGS=$(XWARNINGS) $(OPTIMIZATION) $(XARCH_CFLAGS) $(CPU_CFLAGS) -fno-builtin
+SAFEASMFLAGS=$(XWARNINGS) $(XARCH_CFLAGS) $(CPU_CFLAGS) -fno-builtin
 #
 # on SH4 addition of -g kills the initfini stuff, removed here for all
 # platforms as I suspect no one want initfini compiled -g :-) - davidm
@@ -76,14 +77,14 @@
 	$(AWK) -f $(TOPDIR)extra/scripts/initfini.awk initfini.S
 
 crti.o: crti.S
-	$(CC) $(SAFECFLAGS) -c crti.S -o crti.o
+	$(CC) $(SAFEASMFLAGS) -c crti.S -o crti.o
 
 $(TOPDIR)lib/crti.o: crti.o
 	mkdir -p $(TOPDIR)lib/
 	cp crti.o $(TOPDIR)lib/
 
 crtn.o: crtn.S
-	$(CC) $(SAFECFLAGS) -c crtn.S -o crtn.o
+	$(CC) $(SAFEASMFLAGS) -c crtn.S -o crtn.o
 
 $(TOPDIR)lib/crtn.o: crtn.o
 	mkdir -p $(TOPDIR)lib/
--- uClibc-0.9.20/libc/sysdeps/linux/i386/Makefile.orig	Fri Jun 27 10:23:23 2003
+++ uClibc-0.9.20/libc/sysdeps/linux/i386/Makefile	Thu Jul 24 17:15:50 2003
@@ -46,7 +46,7 @@
 	cp $(CRT0_OBJ) $(TOPDIR)lib/
 
 $(CRT0_OBJ): $(CRT0_SRC)
-	$(CC) $(CFLAGS) -DL_$* $< -c -o $*.o
+	$(CC) -DL_$* $< -c -o $*.o
 	$(STRIPTOOL) -x -R .note -R .comment $*.o
 
 $(SOBJS): %.o : %.S
