--- uClibc-0.9.32/libc/misc/internals/__uClibc_main.c.orig	2011-06-08 21:35:20.000000000 +0200
+++ uClibc-0.9.32/libc/misc/internals/__uClibc_main.c	2011-06-12 09:23:48.324053409 +0200
@@ -118,7 +118,7 @@
  */
 extern int *weak_const_function __errno_location(void);
 extern int *weak_const_function __h_errno_location(void);
-extern void weak_function _stdio_init(void) attribute_hidden;
+extern void weak_function _stdio_init(void);
 #ifdef __UCLIBC_HAS_LOCALE__
 extern void weak_function _locale_init(void) attribute_hidden;
 #endif
--- uClibc-0.9.33/libc/stdlib/_atexit.c.orig	2012-02-01 22:21:04.000000000 +0100
+++ uClibc-0.9.33/libc/stdlib/_atexit.c	2012-02-04 15:34:41.881463595 +0100
@@ -314,7 +314,7 @@
 	return q != 0;
 }
 
-extern void weak_function _stdio_term(void) attribute_hidden;
+extern void weak_function _stdio_term(void);
 attribute_hidden void (*__exit_cleanup)(int) = 0;
 __UCLIBC_MUTEX_INIT(__atexit_lock, PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP);
 
