diff -urN uClibc-0.9.20.org/Makefile uClibc-0.9.20/Makefile
--- uClibc-0.9.20.org/Makefile	2003-08-15 21:45:59.948679264 +0200
+++ uClibc-0.9.20/Makefile	2003-08-15 21:44:17.000000000 +0200
@@ -120,45 +120,31 @@
 
 headers: include/bits/uClibc_config.h
 	rm -f include/asm;
-	@if [ "$(TARGET_ARCH)" = "powerpc" ];then \
-	    ln -fs $(KERNEL_SOURCE)/include/asm-ppc include/asm; \
-	elif [ "$(TARGET_ARCH)" = "mips" ];then \
-	    ln -fs $(KERNEL_SOURCE)/include/asm-mips include/asm; \
-	elif [ "$(TARGET_ARCH)" = "mipsel" ];then \
-	    ln -fs $(KERNEL_SOURCE)/include/asm-mips include/asm; \
+	@if [ "$(TARGET_ARCH)" = "mipsel" ];then \
+	    ln -fs /usr/include/asm include/asm; \
 	    cd $(shell pwd)/libc/sysdeps/linux; \
 	    ln -fs mips mipsel; \
 	    cd $(shell pwd)/ldso/ldso; \
 	    ln -fs mips mipsel; \
 	    cd $(shell pwd)/libpthread/linuxthreads/sysdeps; \
 	    ln -fs mips mipsel; \
-	elif [ "$(TARGET_ARCH)" = "cris" ];then \
-		ln -fs $(KERNEL_SOURCE)/include/asm-cris include/asm; \
 	else \
-	    if [ "$(UCLIBC_HAS_MMU)" != "y" ]; then \
-	    	if [ -d $(KERNEL_SOURCE)/include/asm-$(TARGET_ARCH)nommu ] ; then \
-		    ln -fs $(KERNEL_SOURCE)/include/asm-$(TARGET_ARCH)nommu include/asm;\
-		else \
-		    ln -fs $(KERNEL_SOURCE)/include/asm-$(TARGET_ARCH) include/asm; \
-		fi; \
-	    else \
-		ln -fs $(KERNEL_SOURCE)/include/asm-$(TARGET_ARCH) include/asm; \
-	    fi; \
+	    ln -fs /usr/include/asm include/asm ; \
 	fi;
 	rm -f include/asm-generic;
-	ln -fs $(KERNEL_SOURCE)/include/asm-generic include/asm-generic; 
+	ln -fs /usr/include/asm-generic include/asm-generic; 
 	@if [ ! -f include/asm/unistd.h ] ; then \
 	    set -e; \
 	    echo " "; \
-	    echo "The path '$(KERNEL_SOURCE)/include/asm' doesn't exist."; \
+	    echo "The path '/usr/include/asm' doesn't exist."; \
 	    echo "I bet you did not set KERNEL_SOURCE, TARGET_ARCH or UCLIBC_HAS_MMU"; \
 	    echo "correctly when you configured uClibc.  Please fix these settings."; \
 	    echo " "; \
 	    false; \
 	fi;
 	rm -f include/linux include/scsi
-	ln -fs $(KERNEL_SOURCE)/include/linux include/linux
-	ln -fs $(KERNEL_SOURCE)/include/scsi include/scsi
+	ln -fs /usr/include/linux include/linux
+	ln -fs /usr/include/scsi include/scsi
 	@cd include/bits; \
 	set -e; \
 	for i in `ls ../../libc/sysdeps/linux/common/bits/*.h` ; do \
@@ -259,21 +245,21 @@
 ifeq ($(strip $(HAVE_SHARED)),y)
 	install -d $(PREFIX)$(DEVEL_TOOL_PREFIX)/bin;
 	install -m 755 ldso/util/ldd \
-		$(PREFIX)$(SYSTEM_DEVEL_PREFIX)/bin/$(TARGET_ARCH)-uclibc-ldd
-	ln -fs $(SYSTEM_DEVEL_PREFIX)/bin/$(TARGET_ARCH)-uclibc-ldd \
+		$(PREFIX)$(SYSTEM_DEVEL_PREFIX)/bin/$(TARGET_CPU)-uclibc-ldd
+	ln -fs $(SYSTEM_DEVEL_PREFIX)/bin/$(TARGET_CPU)-uclibc-ldd \
 		$(PREFIX)$(DEVEL_TOOL_PREFIX)/bin/ldd
 	# For now, don't bother with readelf since surely the host
 	# system has binutils, or we couldn't have gotten this far...
 	#install -m 755 ldso/util/readelf \
-	#	$(PREFIX)$(SYSTEM_DEVEL_PREFIX)/bin/$(TARGET_ARCH)-uclibc-readelf
-	#ln -fs $(SYSTEM_DEVEL_PREFIX)/bin/$(TARGET_ARCH)-uclibc-readelf \
+	#	$(PREFIX)$(SYSTEM_DEVEL_PREFIX)/bin/$(TARGET_CPU)-uclibc-readelf
+	#ln -fs $(SYSTEM_DEVEL_PREFIX)/bin/$(TARGET_CPU)-uclibc-readelf \
 	#	$(PREFIX)$(DEVEL_TOOL_PREFIX)/bin/readelf
 	@if [ -x ldso/util/ldconfig ] ; then \
 	    set -x -e; \
 	    install -d $(PREFIX)$(DEVEL_PREFIX)/etc; \
 	    install -m 755 ldso/util/ldconfig \
-		    $(PREFIX)$(SYSTEM_DEVEL_PREFIX)/bin/$(TARGET_ARCH)-uclibc-ldconfig; \
-	    ln -fs $(SYSTEM_DEVEL_PREFIX)/bin/$(TARGET_ARCH)-uclibc-ldconfig \
+		    $(PREFIX)$(SYSTEM_DEVEL_PREFIX)/bin/$(TARGET_CPU)-uclibc-ldconfig; \
+	    ln -fs $(SYSTEM_DEVEL_PREFIX)/bin/$(TARGET_CPU)-uclibc-ldconfig \
 		    $(PREFIX)$(DEVEL_TOOL_PREFIX)/bin/ldconfig; \
 	fi;
 endif
