--- uClibc-0.9.31/libc/misc/internals/__uClibc_main.c.orig	2010-04-02 17:34:27.000000000 +0200
+++ uClibc-0.9.31/libc/misc/internals/__uClibc_main.c	2010-08-19 22:44:43.818819943 +0200
@@ -68,7 +68,7 @@
 extern int *weak_const_function __h_errno_location(void);
 libc_hidden_proto(__h_errno_location)
 
-extern void weak_function _stdio_init(void) attribute_hidden;
+extern void weak_function _stdio_init(void);
 #ifdef __UCLIBC_HAS_LOCALE__
 extern void weak_function _locale_init(void) attribute_hidden;
 #endif
--- uClibc-0.9.29/libc/stdlib/_atexit.c.orig	2007-05-04 16:13:17.000000000 +0200
+++ uClibc-0.9.29/libc/stdlib/_atexit.c	2007-06-04 00:40:04.671116329 +0200
@@ -305,7 +305,7 @@
 #endif
 
 #ifdef L_exit
-extern void weak_function _stdio_term(void) attribute_hidden;
+extern void weak_function _stdio_term(void);
 attribute_hidden void (*__exit_cleanup) (int) = 0;
 __UCLIBC_MUTEX_INIT(__atexit_lock, PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP);
 
