--- uClibc/libc/sysdeps/linux/i386/crti.S	2003/11/05 00:29:20	1.1
+++ uClibc/libc/sysdeps/linux/i386/crti.S	2004/06/12 05:52:22	1.2
@@ -1,40 +1,32 @@
-	.file	"initfini.c"
-#APP
-	
 	.section .init
-#NO_APP
 .globl _init
 	.type	_init, @function
 _init:
 	pushl	%ebp
 	movl	%esp, %ebp
 	pushl	%ebx
-	call	__i686.get_pc_thunk.bx
+	call	__get_pc_thunk_bx
 	addl	$_GLOBAL_OFFSET_TABLE_, %ebx
-#APP
-	
-	
-	
-	
+
+
+
+
 	.section .fini
-#NO_APP
 .globl _fini
 	.type	_fini, @function
 _fini:
 	pushl	%ebp
 	movl	%esp, %ebp
 	pushl	%ebx
-	call	__i686.get_pc_thunk.bx
+	call	__get_pc_thunk_bx
 	addl	$_GLOBAL_OFFSET_TABLE_, %ebx
-#APP
-	
-	
-	
-	.section	.gnu.linkonce.t.__i686.get_pc_thunk.bx,"ax",@progbits
-.globl __i686.get_pc_thunk.bx
-	.hidden	__i686.get_pc_thunk.bx
-	.type	__i686.get_pc_thunk.bx, @function
-__i686.get_pc_thunk.bx:
+
+
+
+	.section	.gnu.linkonce.t.__get_pc_thunk_bx,"ax",@progbits
+.globl __get_pc_thunk_bx
+	.hidden	__get_pc_thunk_bx
+	.type	__get_pc_thunk_bx, @function
+__get_pc_thunk_bx:
 	movl	(%esp), %ebx
 	ret
-	.ident	"GCC: (GNU) 3.3.2 (Debian)"
--- uClibc/libc/sysdeps/linux/i386/crtn.S	2003/11/05 00:29:20	1.1
+++ uClibc/libc/sysdeps/linux/i386/crtn.S	2004/06/12 05:52:22	1.2
@@ -1,33 +1,27 @@
 	.file	"initfini.c"
-#APP
-	
 	.section .init
-#NO_APP
 .globl _init
 	.type	_init, @function
-#NO_APP
 	popl	%ebx
 	popl	%ebp
 	ret
 	.size	_init, .-_init
-#APP
-	
+
+
 	.section .fini
-#NO_APP
 .globl _fini
 	.type	_fini, @function
-#NO_APP
 	popl	%ebx
 	popl	%ebp
 	ret
 	.size	_fini, .-_fini
-#APP
-	
-	.section	.gnu.linkonce.t.__i686.get_pc_thunk.bx,"ax",@progbits
-.globl __i686.get_pc_thunk.bx
-	.hidden	__i686.get_pc_thunk.bx
-	.type	__i686.get_pc_thunk.bx, @function
-__i686.get_pc_thunk.bx:
+
+
+
+	.section	.gnu.linkonce.t.__get_pc_thunk_bx,"ax",@progbits
+.globl __get_pc_thunk_bx
+	.hidden	__get_pc_thunk_bx
+	.type	__get_pc_thunk_bx, @function
+__get_pc_thunk_bx:
 	movl	(%esp), %ebx
 	ret
-	.ident	"GCC: (GNU) 3.3.2 (Debian)"
