--- uClibc-0.9.21/ldso/ldso/sparc/ld_syscalls.h.orig	2002-03-19 10:43:35.000000000 +0000
+++ uClibc-0.9.21/ldso/ldso/sparc/ld_syscalls.h	2003-11-23 20:02:40.000000000 +0000
@@ -8,6 +8,7 @@
 #define __NR_write		  4
 #define __NR_open		  5
 #define __NR_close		  6
+#define __NR_getpid		 20
 #define __NR_getuid		 24
 #define __NR_getgid		 47
 #define __NR_geteuid		 49
--- uClibc-0.9.21/ldso/ldso/sparc/elfinterp.c.orig	2002-11-05 18:21:12.000000000 +0000
+++ uClibc-0.9.21/ldso/ldso/sparc/elfinterp.c	2003-11-23 20:11:55.000000000 +0000
@@ -142,8 +142,8 @@
   return (unsigned int) new_addr;
 }
 
-void _dl_parse_lazy_relocation_information(struct elf_resolve * tpnt, int rel_addr,
-       int rel_size, int type){
+void _dl_parse_lazy_relocation_information(struct elf_resolve * tpnt,
+	   unsigned long rel_addr, unsigned long rel_size, int type){
   int i;
   char * strtab;
   int reloc_type;
@@ -187,8 +187,8 @@
   };
 }
 
-int _dl_parse_relocation_information(struct elf_resolve * tpnt, int rel_addr,
-       int rel_size, int type){
+int _dl_parse_relocation_information(struct elf_resolve * tpnt,
+	   unsigned long rel_addr, unsigned long rel_size, int type){
   int i;
   char * strtab;
   int reloc_type;
@@ -302,8 +302,8 @@
 /* No, there are cases where the SVr4 linker fails to emit COPY relocs
    at all */
 
-int _dl_parse_copy_information(struct dyn_elf * xpnt, int rel_addr,
-       int rel_size, int type)
+int _dl_parse_copy_information(struct dyn_elf * xpnt,
+	   unsigned long rel_addr, unsigned long rel_size, int type)
 {
   int i;
   char * strtab;
--- uClibc-0.9.21/ldso/ldso/ldso.c.orig	2003-09-02 06:55:18.000000000 +0000
+++ uClibc-0.9.21/ldso/ldso/ldso.c	2003-11-23 20:13:09.000000000 +0000
@@ -278,9 +278,9 @@
 #if defined(__i386__)
   __asm__("\tmovl %%ebx,%0\n\t":"=a"(got));
 #elif defined(__m68k__)
-  __asm__("movel %%a5,%0":"=g"(got))
+  __asm__("movel %%a5,%0":"=g"(got));
 #elif defined(__sparc__)
-  __asm__("\tmov %%l7,%0\n\t":"=r"(got))
+  __asm__("\tmov %%l7,%0\n\t":"=r"(got));
 #elif defined(__arm__)
   __asm__("\tmov %0, r10\n\t":"=r"(got));
 #elif defined(__powerpc__)
