diff -urP uClibc-0.9.28/libc/sysdeps/linux/powerpc/__syscall_error.c uClibc-0.9.28-syscallerror/libc/sysdeps/linux/powerpc/__syscall_error.c
--- uClibc-0.9.28/libc/sysdeps/linux/powerpc/__syscall_error.c	2005-08-18 00:49:42.000000000 +0200
+++ uClibc-0.9.28-syscallerror/libc/sysdeps/linux/powerpc/__syscall_error.c	2005-09-19 10:07:47.000000000 +0200
@@ -22,7 +22,7 @@
 
 /* This routine is jumped to by all the syscall handlers, to stash
  * an error number into errno.  */
-int attribute_hidden __syscall_error(int err_no)
+int __syscall_error(int err_no)
 {
 	__set_errno(err_no);
 	return -1;
