--- uClibc-0.9.12/ldso/util/Makefile.orig	Sun Aug 25 17:28:15 2002
+++ uClibc-0.9.12/ldso/util/Makefile	Sun Aug 25 17:39:23 2002
@@ -66,6 +66,7 @@
 	strip -x -R .note -R .comment $@
 
 ldd.target: ldd.c
+	UCLIBC_DEVEL_PREFIX="../../" \
 	$(TARGET_CC) $(CFLAGS) -Wl,-s -DUCLIBC_TARGET_PREFIX=\"$(TARGET_PREFIX)\" \
 		-DUCLIBC_DEVEL_PREFIX=\"$(DEVEL_PREFIX)\" \
 		-DUCLIBC_BUILD_DIR=\"$(shell cd $(TOPDIR) && pwd)\" \
