--- uClibc-0.9.12/extra/gcc-uClibc/Makefile.orig	Thu Jun 20 16:35:19 2002
+++ uClibc-0.9.12/extra/gcc-uClibc/Makefile	Tue Jul 16 12:15:29 2002
@@ -34,7 +34,7 @@
 endif
 
 gcc-uClibc: gcc-uClibc.h gcc-uClibc.c
-	gcc -Wall -O2 -Wl,-s gcc-uClibc.c -o $(TARGET_ARCH)-uclibc-gcc
+	$(NATIVE_CC) -Wall $(NATIVE_CFLAGS) -Wl,-s gcc-uClibc.c -o $(TARGET_ARCH)-uclibc-gcc
 
 ld-uClibc:
 	@echo "#!/bin/sh" > $(TARGET_ARCH)-uclibc-ld
