--- trunk/uClibc/libm/ldouble_wrappers.c	2008/12/22 11:31:12	24487
+++ trunk/uClibc/libm/ldouble_wrappers.c	2009/01/20 00:49:13	24914
@@ -544,7 +544,6 @@
 {
 	return __fpclassify ( (double) x );
 }
-libm_hidden_def(__fpclassifyl)
 #endif
 
 #ifdef L_finitel
@@ -552,7 +551,6 @@
 {
 	return __finite ( (double)x );
 }
-libm_hidden_def(__finitel)
 #endif
 
 #ifdef L_signbitl
@@ -560,7 +558,6 @@
 {
 	return __signbitl ( (double)x );
 }
-libm_hidden_def(__signbitl)
 #endif
 
 #ifdef L_isnanl
@@ -568,7 +565,6 @@
 {
 	return __isnan ( (double)x );
 }
-libm_hidden_def(__isnanl)
 #endif
 
 #ifdef L_isinfl
@@ -576,7 +572,6 @@
 {
 	return __isinf ( (double)x );
 }
-libm_hidden_def(__isinfl)
 #endif
 
 #endif
