diff -ur uClibc/libc/sysdeps/linux/common/syscalls.c
uClibc-setfsuid/sysdeps/linux/common/syscalls.c
--- uClibc/libc/sysdeps/linux/common/syscalls.c	Fri Apr  6 22:28:45 2001
+++ uClibc-setfsuid/sysdeps/linux/common/syscalls.c	Wed May 16 11:14:56 2001
@@ -938,8 +938,16 @@
 //#define __NR_afs_syscall      137
 
 //#define __NR_setfsuid         138
+#ifdef L_setfsuid
+#include <unistd.h>
+_syscall1(int, setfsuid, uid_t, x);
+#endif
 
 //#define __NR_setfsgid         139
+#ifdef L_setfsgid
+#include <unistd.h>
+_syscall1(int, setfsgid, uid_t, x);
+#endif
 
 //#define __NR__llseek          140
 #ifdef L__llseek
