--- trunk/uClibc/libc/sysdeps/linux/common/ppoll.c	2008/11/20 22:11:44	24105
+++ trunk/uClibc/libc/sysdeps/linux/common/ppoll.c	2009/01/28 23:42:01	25109
@@ -17,6 +17,7 @@
    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
    02111-1307 USA.  */
 
+#include <signal.h>
 #include <sys/syscall.h>
 #include <sys/poll.h>
 
@@ -26,24 +27,26 @@
 
 # define __NR___libc_ppoll __NR_ppoll
 static inline
-_syscall4(int, __libc_ppoll, struct pollfd *, fds,
-	nfds_t, nfds, const struct timespec *, timeout,
-	const __sigset_t *, sigmask);
+_syscall5(int, __libc_ppoll,
+	struct pollfd *, fds,
+	nfds_t, nfds,
+	const struct timespec *, timeout,
+	const __sigset_t *, sigmask,
+	size_t, sigsetsize);
 
 int
-ppoll (struct pollfd *fds, nfds_t nfds, const struct timespec *timeout,
+ppoll(struct pollfd *fds, nfds_t nfds, const struct timespec *timeout,
        const __sigset_t *sigmask)
 {
-  /* The Linux kernel can in some situations update the timeout value.
-     We do not want that so use a local variable.  */
-  struct timespec tval;
-  if (timeout != NULL)
-    {
-      tval = *timeout;
-      timeout = &tval;
-    }
+	/* The Linux kernel can in some situations update the timeout value.
+	   We do not want that so use a local variable.  */
+	struct timespec tval;
+	if (timeout != NULL) {
+		tval = *timeout;
+		timeout = &tval;
+	}
 
-  return __libc_ppoll(fds, nfds, timeout, sigmask);
+	return __libc_ppoll(fds, nfds, timeout, sigmask, _NSIG / 8);
 }
 libc_hidden_def(ppoll)
 
