--- uClibc-0.9.26/libc/sysdeps/linux/i386/bits/machine-gmon.h.orig	2003-09-07 05:17:10.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/i386/bits/machine-gmon.h	2004-04-29 09:11:41.487725776 +0200
@@ -30,7 +30,7 @@
 /* We must not pollute the global namespace.  */
 #define mcount_internal __mcount_internal
 
-extern void mcount_internal (u_long frompc, u_long selfpc);
+extern void __attribute((regparm(2))) mcount_internal (u_long frompc, u_long selfpc);
 
 #define _MCOUNT_DECL(frompc, selfpc) \
 void __attribute__ (( regparm (2) )) mcount_internal (u_long frompc, u_long selfpc)
