--- uClibc-0.9.30.2/libc/sysdeps/linux/common/inotify.c~	2010-01-13 22:22:31.000000000 +0100
+++ uClibc-0.9.30.2/libc/sysdeps/linux/common/inotify.c	2010-01-19 17:20:39.994833811 +0100
@@ -15,6 +15,10 @@
 _syscall0(int, inotify_init)
 #endif
 
+#ifdef __NR_inotify_init1
+_syscall1(int, inotify_init1, int, flags)
+#endif
+
 #ifdef __NR_inotify_add_watch
 _syscall3(int, inotify_add_watch, int, fd, const char *, path, uint32_t, mask)
 #endif
--- uClibc-0.9.30.2/libc/sysdeps/linux/common/sys/inotify.h~	2010-01-13 22:22:31.000000000 +0100
+++ uClibc-0.9.30.2/libc/sysdeps/linux/common/sys/inotify.h	2010-01-19 17:21:14.199833529 +0100
@@ -73,6 +73,9 @@
 			  | IN_MOVED_TO | IN_CREATE | IN_DELETE		      \
 			  | IN_DELETE_SELF | IN_MOVE_SELF)
 
+/* Flags for the parameter of inotify_init1.  */
+#define IN_CLOEXEC	02000000
+#define IN_NONBLOCK	04000
 
 __BEGIN_DECLS
 
@@ -79,6 +79,8 @@
 /* Create and initialize inotify instance.  */
 extern int inotify_init (void) __THROW;
 
+extern int inotify_init1 (int flags) __THROW;
+
 /* Add watch of object NAME to inotify instance FD.  Notify about
    events specified by MASK.  */
 extern int inotify_add_watch (int __fd, const char *__name, uint32_t __mask)
