--- uClibc-0.9.32/libpthread/linuxthreads.old/Makefile.in.orig	2011-06-08 21:35:20.000000000 +0200
+++ uClibc-0.9.32/libpthread/linuxthreads.old/Makefile.in	2011-06-12 09:00:50.687340614 +0200
@@ -14,16 +14,16 @@
 CFLAGS-libpthread/linuxthreads.old/sysdeps/$(TARGET_ARCH)/ := $(CFLAGS-linuxthreads.old)
 
 ifeq ($(PTHREADS_DEBUG_SUPPORT),y)
-LDFLAGS-libpthread.so := $(LDFLAGS_NOSTRIP) -Wl,-z,defs
+LDFLAGS-libpthread-uclibc.so := $(LDFLAGS_NOSTRIP) -Wl,-z,defs
 else
-LDFLAGS-libpthread.so := $(LDFLAGS)
+LDFLAGS-libpthread-uclibc.so := $(LDFLAGS)
 endif
-LDFLAGS-$(UCLIBC_FORMAT_DSBT_ELF)-libpthread.so := -Wl,--dsbt-index=10
+LDFLAGS-$(UCLIBC_FORMAT_DSBT_ELF)-libpthread-uclibc.so := -Wl,--dsbt-index=10
 
-LIBS-libpthread.so := $(LIBS) $(ldso)
+LIBS-libpthread-uclibc.so := $(LIBS) $(ldso)
 
-START_FILE-libpthread.so := $(SHARED_START_FILES)
-END_FILE-libpthread.so := $(SHARED_END_FILES)
+START_FILE-libpthread-uclibc.so := $(SHARED_START_FILES)
+END_FILE-libpthread-uclibc.so := $(SHARED_END_FILES)
 
 libpthread_FULL_NAME := libpthread-$(VERSION).so
 
@@ -73,13 +73,13 @@
 libpthread-so-y += $(libpthread_OBJ:.o=.os) $(libpthread-shared-y)
 
 lib-a-$(UCLIBC_HAS_THREADS) += $(top_builddir)lib/libpthread.a
-lib-so-$(UCLIBC_HAS_THREADS) += $(top_builddir)lib/libpthread.so
+lib-so-$(UCLIBC_HAS_THREADS) += $(top_builddir)lib/libpthread-uclibc.so
 
 #ifeq ($(DOMULTI),n)
-$(top_builddir)lib/libpthread.so: $(libpthread_OUT)/libpthread_so.a $(libc.depend)
+$(top_builddir)lib/libpthread-uclibc.so: $(libpthread_OUT)/libpthread_so.a $(libc.depend)
 	$(call link.so,$(libpthread_FULL_NAME),$(ABI_VERSION))
 #else
-#$(top_builddir)lib/libpthread.so: $(libpthread_OUT)/libpthread.oS | $(libc.depend)
+#$(top_builddir)lib/libpthread-uclibc.so: $(libpthread_OUT)/libpthread.oS | $(libc.depend)
 #	$(call linkm.so,$(libpthread_FULL_NAME),$(ABI_VERSION))
 #endif
 
@@ -90,6 +90,10 @@
 	$(Q)$(RM) $@
 	$(do_ar)
 
+$(top_builddir)lib/libpthread.so: $(top_builddir)lib/libpthread-uclibc.so
+	$(LN) -sf $(notdir $<).$(MAJOR_VERSION) $@.$(MAJOR_VERSION)
+	$(LN) -sf $(notdir $<) $@
+
 ifeq ($(PTHREADS_DEBUG_SUPPORT),y)
 $(libpthread_OUT)/libpthread.oS: STRIP_FLAGS:=$(STRIP_FLAGS:-x=-X --strip-debug)
 endif
--- uClibc-0.9.32/libpthread/linuxthreads/Makefile.in.orig	2011-06-08 21:35:20.000000000 +0200
+++ uClibc-0.9.32/libpthread/linuxthreads/Makefile.in	2011-06-12 09:03:14.797345439 +0200
@@ -23,15 +23,15 @@
 #CFLAGS:=$(CFLAGS:-O0=-O1)
 
 ifeq ($(PTHREADS_DEBUG_SUPPORT),y)
-LDFLAGS-libpthread.so := $(LDFLAGS_NOSTRIP) -Wl,-z,defs
+LDFLAGS-libpthread-uclibc.so := $(LDFLAGS_NOSTRIP) -Wl,-z,defs
 else
-LDFLAGS-libpthread.so := $(LDFLAGS)
+LDFLAGS-libpthread-uclibc.so := $(LDFLAGS)
 endif
 
-LIBS-libpthread.so := $(LIBS) $(ldso)
+LIBS-libpthread-uclibc.so := $(LIBS) $(ldso)
 
-START_FILE-libpthread.so := $(SHARED_START_FILES)
-END_FILE-libpthread.so := $(SHARED_END_FILES)
+START_FILE-libpthread-uclibc.so := $(SHARED_START_FILES)
+END_FILE-libpthread-uclibc.so := $(SHARED_END_FILES)
 
 libpthread_FULL_NAME := libpthread-$(VERSION).so
 
@@ -84,13 +84,13 @@
 libpthread-so-y += $(libpthread_OBJ:.o=.oS)
 
 lib-a-$(UCLIBC_HAS_THREADS) += $(top_builddir)lib/libpthread.a
-lib-so-$(UCLIBC_HAS_THREADS) += $(top_builddir)lib/libpthread.so
+lib-so-$(UCLIBC_HAS_THREADS) += $(top_builddir)lib/libpthread-uclibc.so
 
 #ifeq ($(DOMULTI),n)
-$(top_builddir)lib/libpthread.so: $(libpthread_OUT)/libpthread_so.a $(libc.depend) $(top_builddir)lib/libpthread_nonshared.a
+$(top_builddir)lib/libpthread-uclibc.so: $(libpthread_OUT)/libpthread_so.a $(libc.depend) $(top_builddir)lib/libpthread_nonshared.a
 	$(call link.so,$(libpthread_FULL_NAME),$(ABI_VERSION))
 #else
-#$(top_builddir)lib/libpthread.so: $(libpthread_OUT)/libpthread.oS | $(libc.depend) $(top_builddir)lib/libpthread_nonshared.a
+#$(top_builddir)lib/libpthread-uclibc.so: $(libpthread_OUT)/libpthread.oS | $(libc.depend) $(top_builddir)lib/libpthread_nonshared.a
 #	$(call linkm.so,$(libpthread_FULL_NAME),$(ABI_VERSION))
 #endif
 	$(Q)$(RM) $@
@@ -98,6 +98,10 @@
 	$(Q)echo "GROUP ( $(notdir $@).$(ABI_VERSION) libpthread_nonshared.a )" >> $@.tmp
 	$(Q)mv $@.tmp $@
 
+$(top_builddir)lib/libpthread.so: $(top_builddir)lib/libpthread-uclibc.so
+	$(LN) -sf $(notdir $<).$(MAJOR_VERSION) $@.$(MAJOR_VERSION)
+	$(LN) -sf $(notdir $<) $@
+
 ifeq ($(PTHREADS_DEBUG_SUPPORT),y)
 $(libpthread_OUT)/libpthread_so.a: STRIP_FLAGS:=$(STRIP_FLAGS:-x=-X --strip-debug)
 endif
