diff -urN uClibc-0.9.21.org/extra/scripts/fix_includes.sh uClibc-0.9.21/extra/scripts/fix_includes.sh
--- uClibc-0.9.21.org/extra/scripts/fix_includes.sh	2003-09-25 11:01:41.000000000 +0200
+++ uClibc-0.9.21/extra/scripts/fix_includes.sh	2003-09-25 11:05:33.000000000 +0200
@@ -111,47 +111,18 @@
     echo "";
     sleep 1;
 
-    if [ "$TARGET_ARCH" = "powerpc" ];then
-	set -x;
-	ln -fs $KERNEL_SOURCE/include/asm-ppc include/asm;
-	unset -x;
-    elif [ "$TARGET_ARCH" = "mips" ];then
-	set -x;
-	ln -fs $KERNEL_SOURCE/include/asm-mips include/asm;
-	unset -x;
-    elif [ "$TARGET_ARCH" = "arm" ];then
-	set -x;
-	ln -fs $KERNEL_SOURCE/include/asm-arm include/asm;
-	unset -x;
-	if [ ! -L $KERNEL_SOURCE/include/asm-arm/proc ] ; then 
+    set -x;
+    ln -fs $KERNEL_SOURCE/include/asm include/asm;
+    unset -x;
+    
+    if [ "$TARGET_ARCH" = "arm" ];then
+	if [ ! -L $KERNEL_SOURCE/include/asm/proc ] ; then 
 	    if [ ! -L proc ] ; then
 		(cd include/asm;
 		ln -fs proc-armv proc;
 		ln -fs arch-ebsa285 arch);
 	    fi
 	fi;
-    elif [ "$TARGET_ARCH" = "mipsel" ];then
-	set -x;
-	ln -fs $KERNEL_SOURCE/include/asm-mips include/asm;
-	unset -x;
-    elif [ "$TARGET_ARCH)" = "cris" ]; then
-	set -x;
-	ln -fs $KERNEL_SOURCE/include/asm-cris include/asm;
-	unset -x;
-    elif [ "$HAS_MMU" != "y" ]; then
-	    if [ -d $KERNEL_SOURCE/include/asm-${TARGET_ARCH}nommu ] ; then
-		set -x;
-		ln -fs $KERNEL_SOURCE/include/asm-${TARGET_ARCH}nommu include/asm;
-		unset -x;
-	    else
-		set -x;
-		ln -fs $KERNEL_SOURCE/include/asm-$TARGET_ARCH include/asm;
-		unset -x;
-	    fi;
-    else
-	set -x;
-	ln -fs $KERNEL_SOURCE/include/asm-$TARGET_ARCH include/asm;
-	unset -x;
     fi;
 else
 # No guessing required.....
