--- uClibc-0.9.30.3/libc/sysdeps/linux/i386/crtn.S~	2010-03-12 20:32:42.000000000 +0100
+++ uClibc-0.9.30.3/libc/sysdeps/linux/i386/crtn.S	2011-05-27 22:21:25.932407849 +0200
@@ -6,7 +6,6 @@
 	popl	%ebx
 	popl	%ebp
 	ret
-.size _init,.-_init
 
 
 
@@ -16,7 +15,6 @@
 	popl	%ebx
 	popl	%ebp
 	ret
-.size _fini,.-_fini
 
 
 
--- uClibc-0.9.30.3/libc/sysdeps/linux/x86_64/crtn.S~	2010-03-12 20:32:42.000000000 +0100
+++ uClibc-0.9.30.3/libc/sysdeps/linux/x86_64/crtn.S	2011-05-27 22:21:50.250189687 +0200
@@ -7,7 +7,6 @@
 .type    _init, %function
 	addq $8, %rsp
 	ret
-.size _init,.-_init
 
 
 .section .fini
@@ -15,4 +14,3 @@
 .type    _fini, %function
 	addq $8, %rsp
 	ret
-.size _fini, .-_fini
