--- uClibc-0.9.26/extra/Configs/Config.in.orig	2004-01-03 00:34:12.000000000 +0100
+++ uClibc-0.9.26/extra/Configs/Config.in	2004-01-31 17:48:09.355223576 +0100
@@ -1018,7 +1018,7 @@
 
 config RUNTIME_PREFIX
 	string "uClibc runtime library directory"
-	default "/usr/$(TARGET_ARCH)-linux-uclibc/"
+	default "/usr/$(TARGET_CPU)-linux-uclibc"
 	help
 	  RUNTIME_PREFIX is the directory into which the uClibc runtime
 	  libraries will be installed.   The result will look something
@@ -1036,7 +1036,7 @@
 
 config DEVEL_PREFIX
 	string "uClibc development environment directory"
-	default "/usr/$(TARGET_ARCH)-linux-uclibc/usr/"
+	default "/usr/$(TARGET_CPU)-linux-uclibc/usr"
 	help
 	  DEVEL_PREFIX is the directory into which the uClibc development
 	  environment will be installed.   The result will look something
