--- uClibc-0.9.26/libc/sysdeps/linux/alpha/bits/kernel_types.h.orig	2002-08-27 03:20:10.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/alpha/bits/kernel_types.h	2004-01-31 23:16:56.395257648 +0100
@@ -12,7 +12,7 @@
 typedef unsigned int	__kernel_mode_t;
 typedef unsigned int	__kernel_nlink_t;
 typedef long		__kernel_off_t;
-typedef long		__kernel_loff_t;
+typedef long long	__kernel_loff_t;
 typedef int		__kernel_pid_t;
 typedef int		__kernel_ipc_pid_t;
 typedef unsigned int	__kernel_uid_t;
@@ -28,10 +28,13 @@
 typedef unsigned long	__kernel_sigset_t;
 typedef unsigned short	__kernel_uid16_t;
 typedef unsigned short	__kernel_gid16_t;
+typedef int		__kernel_clockid_t;
+typedef int		__kernel_timer_t;
 typedef __kernel_uid_t __kernel_old_uid_t;
 typedef __kernel_gid_t __kernel_old_gid_t;
 typedef __kernel_uid_t __kernel_uid32_t;
 typedef __kernel_gid_t __kernel_gid32_t;
+typedef unsigned int	__kernel_old_dev_t;
 
 typedef struct {
 	int val[2];
--- uClibc-0.9.26/libc/sysdeps/linux/arm/bits/kernel_types.h.orig	2002-08-27 03:20:11.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/arm/bits/kernel_types.h	2004-01-31 23:19:06.474482600 +0100
@@ -22,6 +22,8 @@
 typedef long			__kernel_time_t;
 typedef long			__kernel_suseconds_t;
 typedef long			__kernel_clock_t;
+typedef int			__kernel_timer_t;
+typedef int			__kernel_clockid_t;
 typedef int			__kernel_daddr_t;
 typedef char *			__kernel_caddr_t;
 typedef unsigned short		__kernel_uid16_t;
@@ -30,6 +32,7 @@
 typedef unsigned int		__kernel_gid32_t;
 typedef unsigned short		__kernel_old_uid_t;
 typedef unsigned short		__kernel_old_gid_t;
+typedef unsigned short		__kernel_old_dev_t;
 typedef long long		__kernel_loff_t;
 
 typedef struct {
--- uClibc-0.9.26/libc/sysdeps/linux/cris/bits/kernel_types.h.orig	2002-09-16 10:08:35.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/cris/bits/kernel_types.h	2004-01-31 23:19:59.895361392 +0100
@@ -18,6 +18,8 @@
 typedef long			__kernel_time_t;
 typedef long            __kernel_suseconds_t;
 typedef long			__kernel_clock_t;
+typedef int		__kernel_timer_t;
+typedef int		__kernel_clockid_t;
 typedef int				__kernel_daddr_t;
 typedef char *			__kernel_caddr_t;
 typedef unsigned short  __kernel_uid16_t;
@@ -27,6 +29,7 @@
 
 typedef unsigned short  __kernel_old_uid_t;
 typedef unsigned short  __kernel_old_gid_t;
+typedef unsigned short	__kernel_old_dev_t;
 
 #ifdef __GNUC__
 typedef long long		__kernel_loff_t;
--- uClibc-0.9.26/libc/sysdeps/linux/h8300/bits/kernel_types.h.orig	2002-11-21 07:43:15.000000000 +0100
+++ uClibc-0.9.26/libc/sysdeps/linux/h8300/bits/kernel_types.h	2004-01-31 23:21:02.903782656 +0100
@@ -22,6 +22,8 @@
 typedef long		__kernel_time_t;
 typedef long		__kernel_suseconds_t;
 typedef long		__kernel_clock_t;
+typedef int		__kernel_timer_t;
+typedef int		__kernel_clockid_t;
 typedef int		__kernel_daddr_t;
 typedef char *		__kernel_caddr_t;
 typedef unsigned short	__kernel_uid16_t;
--- uClibc-0.9.26/libc/sysdeps/linux/i386/bits/kernel_types.h.orig	2002-08-27 03:20:12.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/i386/bits/kernel_types.h	2004-01-31 23:21:28.962821080 +0100
@@ -22,6 +22,8 @@
 typedef long		__kernel_time_t;
 typedef long		__kernel_suseconds_t;
 typedef long		__kernel_clock_t;
+typedef int		__kernel_timer_t;
+typedef int		__kernel_clockid_t;
 typedef int		__kernel_daddr_t;
 typedef char *		__kernel_caddr_t;
 typedef unsigned short	__kernel_uid16_t;
@@ -30,6 +32,7 @@
 typedef unsigned int	__kernel_gid32_t;
 typedef unsigned short	__kernel_old_uid_t;
 typedef unsigned short	__kernel_old_gid_t;
+typedef unsigned short	__kernel_old_dev_t;
 typedef long long	__kernel_loff_t;
 
 typedef struct {
--- uClibc-0.9.26/libc/sysdeps/linux/m68k/bits/kernel_types.h.orig	2002-08-27 03:20:13.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/m68k/bits/kernel_types.h	2004-01-31 23:22:01.697844600 +0100
@@ -22,6 +22,8 @@
 typedef long		__kernel_time_t;
 typedef long		__kernel_suseconds_t;
 typedef long		__kernel_clock_t;
+typedef int		__kernel_timer_t;
+typedef int		__kernel_clockid_t;
 typedef int		__kernel_daddr_t;
 typedef char *		__kernel_caddr_t;
 typedef unsigned short	__kernel_uid16_t;
@@ -30,6 +32,7 @@
 typedef unsigned int	__kernel_gid32_t;
 typedef unsigned short	__kernel_old_uid_t;
 typedef unsigned short	__kernel_old_gid_t;
+typedef unsigned short	__kernel_old_dev_t;
 typedef long long	__kernel_loff_t;
 
 typedef struct {
--- uClibc-0.9.26/libc/sysdeps/linux/mips/bits/kernel_types.h.orig	2003-02-23 06:39:26.000000000 +0100
+++ uClibc-0.9.26/libc/sysdeps/linux/mips/bits/kernel_types.h	2004-01-31 23:23:34.192783232 +0100
@@ -58,6 +58,8 @@
 typedef long		__kernel_time_t;
 typedef long		__kernel_suseconds_t;
 typedef long		__kernel_clock_t;
+typedef int		__kernel_timer_t;
+typedef int		__kernel_clockid_t;
 typedef long		__kernel_daddr_t;
 typedef char *		__kernel_caddr_t;
 typedef unsigned short	__kernel_uid16_t;
@@ -66,6 +68,7 @@
 typedef int		__kernel_gid32_t;
 typedef __kernel_uid_t	__kernel_old_uid_t;
 typedef __kernel_gid_t	__kernel_old_gid_t;
+typedef unsigned int	__kernel_old_dev_t;
 typedef long long      __kernel_loff_t;
 #endif
 
--- uClibc-0.9.26/libc/sysdeps/linux/powerpc/bits/kernel_types.h.orig	2002-08-27 03:20:17.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/powerpc/bits/kernel_types.h	2004-01-31 23:24:59.144868552 +0100
@@ -23,6 +23,8 @@
 typedef long		__kernel_ssize_t;
 typedef long		__kernel_ptrdiff_t;
 typedef long		__kernel_time_t;
+typedef int		__kernel_timer_t;
+typedef int		__kernel_clockid_t;
 typedef long		__kernel_suseconds_t;
 typedef long		__kernel_clock_t;
 typedef int		__kernel_daddr_t;
@@ -33,6 +35,7 @@
 typedef unsigned int	__kernel_gid32_t;
 typedef unsigned int	__kernel_old_uid_t;
 typedef unsigned int	__kernel_old_gid_t;
+typedef unsigned long	__kernel_old_dev_t;
 #else
 typedef unsigned int	__kernel_dev_t;
 typedef unsigned int	__kernel_ino_t;
@@ -48,6 +51,8 @@
 typedef long		__kernel_time_t;
 typedef long		__kernel_suseconds_t;
 typedef long		__kernel_clock_t;
+typedef int		__kernel_timer_t;
+typedef int		__kernel_clockid_t;
 typedef int		__kernel_daddr_t;
 typedef char *		__kernel_caddr_t;
 typedef short           __kernel_ipc_pid_t;
@@ -57,6 +62,7 @@
 typedef unsigned int	__kernel_gid32_t;
 typedef unsigned int	__kernel_old_uid_t;
 typedef unsigned int	__kernel_old_gid_t;
+typedef unsigned int	__kernel_old_dev_t;
 typedef long long	__kernel_loff_t;
 #endif
 
--- uClibc-0.9.26/libc/sysdeps/linux/sh/bits/kernel_types.h.orig	2002-08-27 03:20:18.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/sh/bits/kernel_types.h	2004-01-31 23:27:31.495707704 +0100
@@ -22,6 +22,8 @@
 typedef long		__kernel_time_t;
 typedef long		__kernel_suseconds_t;
 typedef long		__kernel_clock_t;
+typedef int		__kernel_timer_t;
+typedef int		__kernel_clockid_t;
 typedef int		__kernel_daddr_t;
 typedef char *		__kernel_caddr_t;
 typedef unsigned short	__kernel_uid16_t;
@@ -30,6 +32,7 @@
 typedef unsigned int	__kernel_gid32_t;
 typedef unsigned short	__kernel_old_uid_t;
 typedef unsigned short	__kernel_old_gid_t;
+typedef unsigned short	__kernel_old_dev_t;
 typedef long long	__kernel_loff_t;
 
 typedef struct {
--- uClibc-0.9.26/libc/sysdeps/linux/sparc/bits/kernel_types.h.orig	2002-08-27 03:20:20.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/sparc/bits/kernel_types.h	2004-01-31 23:28:58.768440232 +0100
@@ -28,10 +28,13 @@
 typedef char *                 __kernel_caddr_t;
 typedef unsigned short	       __kernel_uid16_t;
 typedef unsigned short	       __kernel_gid16_t;
+typedef int                    __kernel_clockid_t;
+typedef int                    __kernel_timer_t;
 typedef __kernel_uid_t 	       __kernel_old_uid_t;
 typedef __kernel_gid_t         __kernel_old_gid_t;
 typedef __kernel_uid_t	       __kernel_uid32_t;
 typedef __kernel_gid_t	       __kernel_gid32_t;
+typedef unsigned int	       __kernel_old_dev_t;
 typedef int		       __kernel_suseconds_t;
 typedef long long              __kernel_loff_t;
 #else
@@ -59,6 +62,9 @@
 typedef unsigned int	       __kernel_gid32_t;
 typedef unsigned short	       __kernel_old_uid_t;
 typedef unsigned short	       __kernel_old_gid_t;
+typedef unsigned short	       __kernel_old_dev_t;
+typedef int                    __kernel_clockid_t;
+typedef int                    __kernel_timer_t;
 typedef long long              __kernel_loff_t;
 #endif
 
--- uClibc-0.9.26/libc/sysdeps/linux/v850/bits/kernel_types.h.orig	2003-01-31 10:20:24.000000000 +0100
+++ uClibc-0.9.26/libc/sysdeps/linux/v850/bits/kernel_types.h	2004-01-31 23:29:54.212011528 +0100
@@ -31,6 +31,8 @@
 typedef long		__kernel_time_t;
 typedef long		__kernel_suseconds_t;
 typedef long		__kernel_clock_t;
+typedef int		__kernel_timer_t;
+typedef int		__kernel_clockid_t;
 typedef int		__kernel_daddr_t;
 typedef char *		__kernel_caddr_t;
 typedef unsigned short	__kernel_uid16_t;
@@ -40,6 +42,7 @@
 
 typedef unsigned short	__kernel_old_uid_t;
 typedef unsigned short	__kernel_old_gid_t;
+typedef unsigned int	__kernel_old_dev_t;
 
 typedef struct {
 #ifdef __USE_ALL
