diff -ur uClibc/include/features.h uClibc-/include/features.h
--- uClibc/include/features.h	Wed Nov 21 15:56:40 2001
+++ uClibc-/include/features.h	Tue Jan 22 11:19:35 2002
@@ -327,6 +327,12 @@
 #include <bits/uClibc_config.h>
 #undef __need_uClibc_config_h
 
+#ifndef __UCLIBC_HAVE_LFS__
+# undef __USE_LARGEFILE
+# undef __USE_LARGEFILE64
+# undef __USE_FILE_OFFSET64
+#endif
+
 /* Some nice features only work properly with ELF */
 #if defined HAVE_ELF	
 #   define link_warning(symbol, msg)					      \
