--- uClibc-0.9.27/Rules.mak.orig	2005-01-12 08:59:21.000000000 +0100
+++ uClibc-0.9.27/Rules.mak	2005-01-15 20:27:18.203377192 +0100
@@ -285,7 +285,8 @@
     CFLAGS += $(PICFLAG)
 endif
 
-ASFLAGS = $(CFLAGS)
+ASFLAGS = $(XWARNINGS) $(XARCH_CFLAGS) $(CPU_CFLAGS) $(SSP_CFLAGS) -fno-builtin -nostdinc -D_LIBC -I$(TOPDIR)include -I. -iwithprefix include
+
 ifeq ($(UCLIBC_BUILD_NOEXECSTACK),y)
 check_as_noexecstack=$(shell if $(LD) --help | grep -q "z noexecstack"; then echo "-Wa,--noexecstack"; fi)
 ASFLAGS += $(check_as_noexecstack)
