diff -urN uClibc-0.9.26.org/libpthread/Makefile uClibc-0.9.26/libpthread/Makefile
--- uClibc-0.9.26.org/libpthread/Makefile	2004-01-25 15:20:49.547594064 +0100
+++ uClibc-0.9.26/libpthread/Makefile	2004-01-25 15:33:34.026392026 +0100
@@ -21,7 +21,8 @@
 
 #Adjust the soname version to avoid namespace collisions with glibc's libpthread
 LIBPTHREAD=libpthread.a
-LIBPTHREAD_SHARED=libpthread.so
+LIBPTHREAD_SHARED=libpthread-uclibc.so
+LIBPTHREAD_SHARED_ORG=libpthread.so
 LIBPTHREAD_SHARED_FULLNAME=libpthread-$(MAJOR_VERSION).$(MINOR_VERSION).$(SUBLEVEL).so
 
 LIBTHREAD_DB=libthread_db.a
@@ -90,6 +91,8 @@
 			$(TOPDIR)lib/$(LIBPTHREAD_SHARED); \
 		$(LN) -sf $(LIBPTHREAD_SHARED_FULLNAME) \
 			$(TOPDIR)lib/$(LIBPTHREAD_SHARED).$(MAJOR_VERSION); \
+		$(LN) -sf $(LIBPTHREAD_SHARED_FULLNAME) \
+			$(TOPDIR)lib/$(LIBPTHREAD_SHARED_ORG).$(MAJOR_VERSION); \
 	fi;
 	if [ -f $(LIBTHREAD_DB) ] ; then \
 		set -e; \
